processor.go

```diff
@@ -122,9 +122,13 @@ func NewProcessor(client *Client, subscriptionID string, options *ProcessorOptio
 			InitialRetryInterval: options.InitialRetryInterval,
 			MaxRetryInterval:     options.MaxRetryInterval,
 			CommitMaxElapsedTime: options.CommitMaxElapsedTime,
-			NotifyErr:            func(err error, duration time.Duration) { options.NotifyErr(i, err, duration) },
-			NotifyOK:             func() { options.NotifyOK(i) }}
-
+			NotifyErr: func(streamNo uint) func(err error, duration time.Duration) {
```

Comment:
`minor` you can achieve capture of the value in the closure just by assigning `i` to a local variable `streamNo` inside of the loop:
```go
for i := uint(0); i < options.StreamCount; i++ {
	streamNo := i
	streamOptions := StreamOptions{
		BatchLimit:           options.BatchLimit,
		FlushTimeout:         options.FlushTimeout,
		MaxUncommittedEvents: options.MaxUncommittedEvents,
		InitialRetryInterval: options.InitialRetryInterval,
		MaxRetryInterval:     options.MaxRetryInterval,
		CommitMaxElapsedTime: options.CommitMaxElapsedTime,
		NotifyErr: func(err error, duration time.Duration) { options.NotifyErr(streamNo, err, duration) },
		NotifyOK: func() { options.NotifyOK(streamNo) },
	}
	processor.streamOptions = append(processor.streamOptions, streamOptions)
}
```
Unless I'm missing something here, I think I prefer the version using the local variable in this case. What do you think?

Reply:
fixed